src/Form/EventListener/AnyNoneFilterListener.php

```diff
@@ -45,26 +48,46 @@ class AnyNoneFilterListener
      */
     protected $columnsExtractor;
 
+    /**
+     * Elements (listing entities) containing AnyNoneSelection
+     *
+     * @var string[]
+     */
+    private $entityElementsWithAnyNone = array();
+
     public function __construct(ColumnsExtractor $columnsExtractor)
     {
         $this->columnsExtractor = $columnsExtractor;
     }
 
+    /**
+     * Adds data for AnyNoneFilter to the form (events, fields, ...).
+     *
+     * @param \CubeTools\CubeCommonBundle\Form\EventListener\FormBuilderInterface $builder
+     */
+    public function addToBuilder(FormBuilderInterface $builder)
+    {
+        $this->addAnyNoneFields($builder);
+        $builder->addEventListener(FormEvents::PRE_SUBMIT, array($this, 'processAnyNoneColumns'));
+    }
+
     /**
      * Method to be called as PRE_SET_DATA listener for form.
      *
+     * It is recommended to use {@see addToBuilder()} instead of registering the events.
+     *
+     * Add fields to the form to submit data to frontend.
+     *
      * @param \Symfony\Component\Form\FormEvent $event
      */
     public function addAnyNoneColumns(FormEvent $event)
     {
-        $event->getForm()->add(self::KEY_ANY_NONE_COLUMNS, HiddenType::class, array(
-            'data' => json_encode($this->columnsExtractor->getEntitiesColumnsByName($event->getForm())),
-        ));
-        $event->getForm()->add(self::KEY_ANY_NONE_SELECTED_COLUMNS, HiddenType::class);
+        $this->addAnyNoneFields($event->getForm());
```

Comment:
commented here: https://github.com/EmchBerger/cube-common-bundle/pull/113/files#r239070035

Reply:
agree, only makes sense in this PR, but not in the other

Reply:
sorry, makes sense here because used in addToBuilder()

Reply:
private method is deleted now